mondrian/src/main/java/mondrian/spi/DynamicSchemaProcessor.java

```diff
@@ -50,9 +50,20 @@
      * @return Returns the modified schema.
      * @throws Exception if an error occurs.
      */
-    public String processSchema(
-        String schemaUrl,
-        Util.PropertyList connectInfo) throws Exception;
+    public String processSchema( String schemaUrl, Util.PropertyList connectInfo) throws Exception;
+
+
+    /**
+     * Modifies a Mondrian catalog.
+     *
+     * <p>An implementation should process catalog and apply dynamic changes
+     *
+     * @param catalog The URL of the catalog.
+     * @param connectInfo Connection properties.
+     * @return Returns the modified schema.
+     * @throws Exception if an error occurs.
+     */
+    public String processCatalog( String catalog, Util.PropertyList connectInfo) throws Exception;
```

Comment:
This method needs a default implementation in the interface. If not it will break all of the integrations by customers :)

https://www.baeldung.com/java-static-default-methods

The default should just return the catalog string as-is. It will also allow you to remove all of the new methods you put in the test classes.